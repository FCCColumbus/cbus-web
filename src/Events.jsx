// import Event from './Event'

function Events() {
    // The arrays below are retained for future reference,
    // pending implementation of some API call to retrieve events from Google Calendar or Meetup.com, for example.

    // const events = [{
    //     name: "regular meetup",
    //     description: "have pizza and talk about code",
    //     time: new Date('2023-04-17T18:30:00'),
    //     location: 'Co-Hatch Upper Arlington',
    //     id: 'event-01'
    // }, {
    //     name: "What are web components?",
    //     description: "John will give a lecture about what web components are and why we should care. Don't worry, there'll be pizza!",
    //     time: new Date('2023-05-17T18:30:00'),
    //     location: 'Co-Hatch Upper Arlington',
    //     id: 'event-02'
    // }]

    return (
        <>
            <div id="events-section" className="events">
                <h3>Upcoming Events</h3>
                <div id="events-placeholder-text">
                    The &quot;Events&quot; section is under development.
                    <br /><br />
                    In the meantime, please visit
                    <br />
                    <a href="https://www.meetup.com/techlifecolumbus/events/">https://www.meetup.com/techlifecolumbus/events/</a>
                    <br />
                    and look for FreeCodeCamp Columbus events there!
                </div>
                {/* The array mapping below is retained for future reference. */}
                {/* {events.map((event, i) => <Event event={event} key={event.id} id={i} />)} */}
            </div>
            <img src='/divider.jpg' alt='code on a screen. mostly css' />
        </>
    );
}

export default Events;
