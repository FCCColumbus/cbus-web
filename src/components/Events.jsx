import Calendar from 'react-calendar';
import { useState, useEffect, useMemo } from 'react';
import ical from 'cal-parser';

import Event from './Event';

import 'react-calendar/dist/Calendar.css';

function Events() {
  const [selectedDate, setSelectedDate] = useState(new Date());
  const [events, setEvents] = useState([]);
  const [loaded, setLoaded] = useState(false);

  const dateOnly = (date) =>
    new Date(date.getFullYear(), date.getMonth(), date.getDate());

  const eventsByDateString = useMemo(() => {
    // Key: date string of date object
    // Value: Array of event objects
    const eventsDict = {};
    events.forEach((event) => {
      const dateString = event.dtstart.value.toDateString();
      if (!eventsDict[dateString]) {
        eventsDict[dateString] = [];
      }
      eventsDict[dateString].push(event);
    });
    return eventsDict;
  }, [events]);

  const displayedEvents = useMemo(
    () => eventsByDateString[selectedDate.toDateString()] ?? [],
    [eventsByDateString, selectedDate]
  );

  const fetchData = async () => {
    try {
      const response = await fetch(
        'https://docs.google.com/document/d/1OcnWWt1qHaJWE-8v_FPtNO8DKviRQ7DMaqylupuvPXE/export?format=txt'
      );

      const content = await response.text();
      return content;
    } catch (error) {
      return null;
    }
  };

  useEffect(() => {
    const eventsData = async () => {
      try {
        const fetchedData = await fetchData();
        const parsedData = ical.parseString(fetchedData);
        return parsedData.events;
      } catch (error) {
        return [];
      }
    };
    eventsData().then((data) => {
      setEvents(data);
      if (data.length > 0) {
        setLoaded(true);
      }
    });
  }, []);

  useEffect(() => {
    const initialSelectedDate = () => {
      // Assumes events is sorted chronologically.
      // Returns today's date if there is an event.
      // If not, returns the nearest future date with an event.
      // If no future events, returns the last date with an event.

      const currentDate = dateOnly(new Date());

      if (events.length === 0) {
        return currentDate;
      }
      // Check if today's date has an event
      if (eventsByDateString[currentDate.toDateString()]) {
        return currentDate;
      }

      // Check if today's date is after the last event date
      const lastEventDate = dateOnly(events[events.length - 1].dtstart.value);
      if (lastEventDate.getTime() < currentDate.getTime()) {
        return lastEventDate;
      }

      // Binary search for nearest future date with an event
      // when there's no event today
      let left = 0;
      let right = events.length - 1;
      while (left < right) {
        const mid = Math.floor((left + right) / 2);
        const midDate = dateOnly(events[mid].dtstart.value);

        if (midDate.getTime() > currentDate.getTime()) {
          right = mid;
        } else {
          left = mid + 1;
        }
      }
      return dateOnly(events[left].dtstart.value);
    };
    setSelectedDate(initialSelectedDate());
  }, [events, eventsByDateString]);

  return (
    <div id="events" className="events">
      <h3>Upcoming Events</h3>
      <div id="events-placeholder-text">
        The &quot;Events&quot; section is under development.
        <br />
        <br />
        In the meantime, please visit
        <br />
        <a href="https://www.meetup.com/techlifecolumbus/events/">
          https://www.meetup.com/techlifecolumbus/events/
        </a>
        <br />
        and look for FreeCodeCamp Columbus events there!
      </div>
      {loaded && (
        <div className="calendar-events-container">
          <Calendar
            className="calendar"
            onChange={setSelectedDate}
            value={selectedDate}
            locale="en-US"
            minDetail="year"
            aria-label="Event Calendar"
            nextAriaLabel="Next"
            prevAriaLabel="Previous"
            next2AriaLabel="Jump forwards"
            prev2AriaLabel="Jump backwards"
            tileClassName={({ date, view }) =>
              view === 'month' && eventsByDateString[date.toDateString()]
                ? 'event-tile'
                : ''
            }
          />
          <div
            className={`event-container ${
              displayedEvents.length > 1 ? 'multiple-events' : ''
            }`}
          >
            {displayedEvents.map((event) => (
              <Event key={event.uid.value} event={event} />
            ))}
          </div>
        </div>
      )}
    </div>
  );
}

export default Events;
