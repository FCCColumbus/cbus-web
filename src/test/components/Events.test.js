import renderer from "react-test-renderer";

import Events from "../../Events";

describe("Events", () => {
    it("renders", () => {
        const view = renderer.create(<Events />);
        expect(view).toMatchSnapshot();
    });
});
