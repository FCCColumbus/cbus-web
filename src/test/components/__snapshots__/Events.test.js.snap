// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders 1`] = `
<div
  className="events"
  id="events"
>
  <h3>
    Upcoming Events
  </h3>
  <div
    id="events-placeholder-text"
  >
    The "Events" section is under development.
    <br />
    <br />
    In the meantime, please visit
    <br />
    <a
      href="https://www.meetup.com/techlifecolumbus/events/"
    >
      https://www.meetup.com/techlifecolumbus/events/
    </a>
    <br />
    and look for FreeCodeCamp Columbus events there!
  </div>
  <div
    className="calendar-events-container"
  >
    <div
      className="react-calendar calendar"
    >
      <div
        className="react-calendar__navigation"
      >
        <button
          aria-label="Jump backwards"
          className="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          «
        </button>
        <button
          aria-label="Previous"
          className="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          ‹
        </button>
        <button
          aria-label=""
          className="react-calendar__navigation__label"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "flexGrow": 1,
            }
          }
          type="button"
        >
          <span
            className="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
          >
            March 2024
          </span>
        </button>
        <button
          aria-label="Next"
          className="react-calendar__navigation__arrow react-calendar__navigation__next-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          ›
        </button>
        <button
          aria-label="Jump forwards"
          className="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          »
        </button>
      </div>
      <div
        className="react-calendar__viewContainer"
      >
        <div
          className="react-calendar__month-view"
        >
          <div
            style={
              Object {
                "alignItems": "flex-end",
                "display": "flex",
              }
            }
          >
            <div
              style={
                Object {
                  "flexGrow": 1,
                  "width": "100%",
                }
              }
            >
              <div
                className="react-calendar__month-view__weekdays"
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": undefined,
                    "flexWrap": "nowrap",
                  }
                }
              >
                <div
                  className="react-calendar__month-view__weekdays__weekday"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Monday"
                    title="Monday"
                  >
                    Mon
                  </abbr>
                </div>
                <div
                  className="react-calendar__month-view__weekdays__weekday"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Tuesday"
                    title="Tuesday"
                  >
                    Tue
                  </abbr>
                </div>
                <div
                  className="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--current"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Wednesday"
                    title="Wednesday"
                  >
                    Wed
                  </abbr>
                </div>
                <div
                  className="react-calendar__month-view__weekdays__weekday"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Thursday"
                    title="Thursday"
                  >
                    Thu
                  </abbr>
                </div>
                <div
                  className="react-calendar__month-view__weekdays__weekday"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Friday"
                    title="Friday"
                  >
                    Fri
                  </abbr>
                </div>
                <div
                  className="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Saturday"
                    title="Saturday"
                  >
                    Sat
                  </abbr>
                </div>
                <div
                  className="react-calendar__month-view__weekdays__weekday react-calendar__month-view__weekdays__weekday--weekend"
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                >
                  <abbr
                    aria-label="Sunday"
                    title="Sunday"
                  >
                    Sun
                  </abbr>
                </div>
              </div>
              <div
                className="react-calendar__month-view__days"
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": undefined,
                    "flexWrap": "wrap",
                  }
                }
              >
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="February 26, 2024"
                  >
                    26
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="February 27, 2024"
                  >
                    27
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="February 28, 2024"
                  >
                    28
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="February 29, 2024"
                  >
                    29
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 1, 2024"
                  >
                    1
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 2, 2024"
                  >
                    2
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 3, 2024"
                  >
                    3
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 4, 2024"
                  >
                    4
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 5, 2024"
                  >
                    5
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 6, 2024"
                  >
                    6
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 7, 2024"
                  >
                    7
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 8, 2024"
                  >
                    8
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 9, 2024"
                  >
                    9
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 10, 2024"
                  >
                    10
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 11, 2024"
                  >
                    11
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 12, 2024"
                  >
                    12
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 13, 2024"
                  >
                    13
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 14, 2024"
                  >
                    14
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 15, 2024"
                  >
                    15
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 16, 2024"
                  >
                    16
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 17, 2024"
                  >
                    17
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 18, 2024"
                  >
                    18
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 19, 2024"
                  >
                    19
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__tile--now react-calendar__tile--active react-calendar__tile--range react-calendar__tile--rangeStart react-calendar__tile--rangeEnd react-calendar__tile--rangeBothEnds react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 20, 2024"
                  >
                    20
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 21, 2024"
                  >
                    21
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 22, 2024"
                  >
                    22
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 23, 2024"
                  >
                    23
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 24, 2024"
                  >
                    24
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 25, 2024"
                  >
                    25
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 26, 2024"
                  >
                    26
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 27, 2024"
                  >
                    27
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 28, 2024"
                  >
                    28
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 29, 2024"
                  >
                    29
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 30, 2024"
                  >
                    30
                  </abbr>
                </button>
                <button
                  className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                  onClick={[Function]}
                  style={
                    Object {
                      "flexBasis": "14.285714285714286%",
                      "flexGrow": 0,
                      "flexShrink": 0,
                      "marginInlineEnd": 0,
                      "marginInlineStart": null,
                      "marginLeft": null,
                      "overflow": "hidden",
                    }
                  }
                  type="button"
                >
                  <abbr
                    aria-label="March 31, 2024"
                  >
                    31
                  </abbr>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="event-container "
    />
  </div>
</div>
`;
